app/pics_applicable_test_cases.py

```diff
@@ -40,15 +43,37 @@ def applicable_test_cases_list(pics: PICS) -> PICSApplicableTestCases:
     test_collections = test_script_manager.test_collections
     enabled_pics = set([item.number for item in pics.all_enabled_items()])
 
-    for test_collection in test_collections.values():
-        for test_suite in test_collection.test_suites.values():
-            for test_case in test_suite.test_cases.values():
-                if len(test_case.pics) == 0:
-                    # Test cases without pics required are always applicable
-                    applicable_tests.add(test_case.metadata["title"])
-                elif len(test_case.pics) > 0:
-                    if test_case.pics.issubset(enabled_pics):
-                        applicable_tests.add(test_case.metadata["title"])
+    applicable_mandatories_tests = __append_test_cases(
+        test_collections, enabled_pics, True
+    )
+    applicable_remaining_tests = __append_test_cases(
+        test_collections, enabled_pics, False
+    )
+
+    # Add first the mandatories test cases
+    applicable_tests.extend(applicable_mandatories_tests)
+    # Add the remaining test cases
+    applicable_tests.extend(applicable_remaining_tests)
 
     logger.debug(f"Applicable test cases: {applicable_tests}")
     return PICSApplicableTestCases(test_cases=applicable_tests)
+
+
+def __append_test_cases(
```

Comment:
I think `__append_test_cases` suggests that the function will modify some list by appending the test cases, but it doesn't modify an existing object, it returns a new list.
```suggestion
def __applicable_test_cases(
```